From 6fa79f50a9d4c3a565a68c4a87b0f29686da6553 Mon Sep 17 00:00:00 2001
From: Huang Jiachai <hjc@rock-chips.com>
Date: Thu, 10 Mar 2016 10:05:19 +0800
Subject: [PATCH] video: rockchip: lcdc: 3366: fix timing reg take effect time

rk3366 timing reg config change to frame effect,
so we need config done after update timing.

Change-Id: I7279fc03a066357cb8a0ed452e9182f92bf90f01
Signed-off-by: Huang Jiachai <hjc@rock-chips.com>
---
 drivers/video/rockchip/lcdc/rk3368_lcdc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/lcdc/rk3368_lcdc.c b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
index a1786d5346db..52387869001f 100644
--- a/drivers/video/rockchip/lcdc/rk3368_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk3368_lcdc.c
@@ -2163,6 +2163,8 @@ static int rk3368_load_screen(struct rk_lcdc_driver *dev_drv, bool initscreen)
 				     v_DSP_LUT_EN(1));
 		rk3368_lcdc_bcsh_path_sel(dev_drv);
 		rk3368_config_timing(dev_drv);
+		if (lcdc_dev->soc_type == VOP_FULL_RK3366)
+			lcdc_cfg_done(lcdc_dev);
 	}
 	spin_unlock(&lcdc_dev->reg_lock);
 	rk3368_lcdc_set_dclk(dev_drv, 1);
-- 
2.35.3

